File: ParadigmaComp/15Questoes/python/questao3.py
# Encapsulamento com Classe ContaBancaria
class ContaBancaria:
    def __init__(self, titular, saldo=0):
        self.__saldo = saldo
        self.titular = titular

    def depositar(self, valor):
        self.__saldo += valor

    def sacar(self, valor):
        if valor <= self.__saldo:
            self.__saldo -= valor
        else:
            print("Saldo insuficiente.")

    def exibir_saldo(self):
        return f"Saldo: {self.__saldo}"

# Exemplo de uso
conta = ContaBancaria("João")
conta.depositar(1000)
print(conta.exibir_saldo())
conta.sacar(500)
print(conta.exibir_saldo())
conta.sacar(600)

File: ParadigmaComp/15Questoes/golang/questao3.go
package main

import "fmt"

// Encapsulamento com Classe ContaBancaria

type ContaBancaria struct {
	titular string
	saldo   float64
}

func (c *ContaBancaria) Depositar(valor float64) {
	c.saldo += valor
}

func (c *ContaBancaria) Sacar(valor float64) {
	if valor <= c.saldo {
		c.saldo -= valor
	} else {
		fmt.Println("Saldo insuficiente.")
	}
}

func (c ContaBancaria) ExibirSaldo() string {
	return fmt.Sprintf("Saldo: %.2f", c.saldo)
}

func main() {
	conta := ContaBancaria{titular: "João", saldo: 0}
	conta.Depositar(1000)
	fmt.Println(conta.ExibirSaldo())
	conta.Sacar(500)
	fmt.Println(conta.ExibirSaldo())
	conta.Sacar(600)
}
